doc/changelog/03-notations/16788-fix_16785.rst

```diff
@@ -0,0 +1,9 @@
+- **Changed:**
+  Don't make `\`[=\`` a keyword. Users of the `\`[=\``
+  token in a notation should replace it with the pair of
+  tokens `\`[\` \`=\``. If compatibility with Coq < 8.17
+  is needed, the given notations can be written with
+  a space (`[ =` instead of `[=`) when used
```

Comment:
```suggestion
- **Removed:**
  The `\'[=\'` keyword. `\'[=\'`
  tokens in notation definitions should be replaced with the pair of
  tokens `\'[\' \'=\'`. If compatibility with Coq <= 8.17
  is needed, replace `[=` in uses of the notation with
  an added space (`[ =`)
```

This is a removal.  It is the users who need to make changes :-).
I expect it's <= 8.17, right?
Please be clear whether you're referring to _uses_ of the notation or the _definition_.
Note that you used an accent ``` ` ``` where apostrophes (`'`) would be used, but it's not clear to me that `Notation` requires apostrophes around `[=` `[` or `=`.  If not needed, please remove the apostrophes.

It would be nice to verify this renders correctly by locally copying the change log into an existing rst of the doc, rebuilding the manual and inspecting for correct appearance.

Reply:
Your original text renders like this:

![image](https://user-images.githubusercontent.com/1253341/219107861-c6996e4c-a7ab-4f40-9569-ae8f03cb7e08.png)

Reply:
> This is a removal. It is the users who need to make changes :-).

Indeed, hence my initial choice for "changed". But you're right it's rather a removal.

> I expect it's <= 8.17, right?

Indeed, thanks! (I was initially targetting 8.17 months ago and forgot to update)

> Please be clear whether you're referring to _uses_ of the notation or the _definition_.

Right, thanks for your suggestion, it's much better.

> it's not clear to me that `Notation` requires apostrophes around `[=` `[` or `=`. If not needed, please remove the apostrophes.

They are not strictly required but I feel they help the reading here. I might very well be wrong, feel free to disagree.

> It would be nice to verify this renders correctly by locally copying the change log into an existing rst of the doc, rebuilding the manual and inspecting for correct appearance.

I wouldn't say the result looks nice, but it seems correct to me.